core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -76,6 +93,16 @@ oppia.directive('collectionNodeCreator', [
             );
           };
 
+          var convertTypeAheadToExplorationId = function(newExplorationId) {
+            if (newExplorationId.length > 1 && newExplorationId[0] === '(') {
+              var explorationId = '';
+              for (var i = 2; newExplorationId[i] !== ')'; i++) {
+                explorationId += newExplorationId[i];
+              }
+              return explorationId;
+            }
+          };
```

Comment:
I've updated the code and I do not expect a failure. It either returns the match from the pattern (id+exploration title) or the exploration id from the input.